packages/typegpu/src/tgpuFn.ts

```diff
@@ -204,27 +210,32 @@ function createFn<
         externalMap[key] = value;
 
         // Giving name to external value
-        if (
-          isNamable(value) &&
-          (!('label' in value) || value.label === undefined)
-        ) {
+        if (isNamable(value)) {
           value.$name(key);
         }
       }
       return this;
     },
 
+    $__ast(argNames: string[], body: Block): TgpuTgslFnBase<Args, Return> {
+      prebuiltAst = { argNames, body };
+      return this;
+    },
+
     $name(newLabel: string): TgpuTgslFnBase<Args, Return> {
-      // @ts-ignore
-      this.label = newLabel;
+      // Can only set the name once, other calls are ignored.
```

Comment:
won't this be quite confusing for the users? case: they import our cli-generated function which we will name automatically, then they run $name with a new name, but it will not apply